sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -333,10 +333,11 @@ case class AlterTableUnsetPropertiesCommand(
     val catalog = sparkSession.sessionState.catalog
     val table = catalog.getTableRawMetadata(tableName)
     if (!ifExists) {
-      propKeys.foreach { k =>
-        if (!table.properties.contains(k) && k != TableCatalog.PROP_COMMENT) {
-          throw QueryCompilationErrors.unsetNonExistentPropertyError(k, table.identifier)
-        }
+      val nonexistentKeys = propKeys.filter(key => !table.properties.contains(key)
```

Comment:
I think instead of printing the first nonexistent property in the error message, it is better to collect all the properties which are not existent and print them all.